### PR TITLE
Integration Test Upgrade for Node 18

### PR DESCRIPTION
**Pull request by charleshu-8**
- Part of Node 18 upgrade piecemealing
- Attempt to solve Cypress test unreliability

### CODE REVIEW

**Review by Amndeep7 — CHANGES_REQUESTED** [review]
i understand that the best practice is to not use async await for the callback functions for the tests themselves, which is why eslint is yelling at us.  however, i think that's cause cypress is expecting us to handle asynchronous stuff through other paradigms.  try out some of the solutions in this stackoverflow question and see what works / feels good to write / looks aesthetic: https://stackoverflow.com/questions/49980311/cypress-io-how-to-handle-async-code.  after we can get it working in the individual tests themselves, let's see if we can move stuff back out into the beforeeach.

File: test/support/index.ts [hunk 1]
```diff
@@ -26,6 +26,11 @@ Cypress.on('uncaught:exception', (err) => {
   return !err.message.includes('getBoundingClientRect');
 });
 
+// Extends timeout counter to 30000 ms
+Cypress.config('defaultCommandTimeout', 30000);
```
Comment (Amndeep7) [inline test/support/index.ts:30]: shouldn't we change the config here? https://github.com/mitre/heimdall2/blob/master/cypress.config.ts

**Review by Amndeep7 — CHANGES_REQUESTED** [review]

File: test/support/index.ts (hunk 1 above)
Comment (charleshu-8) [reply test/support/index.ts]: Changed as requested.

**Review by Amndeep7 — APPROVED** [review]

**Review by Amndeep7 — APPROVED** [review]
Let's get this merged in.  IIRC when testing with node18, it might be necessary to do the Henry/Charles's original approach where we have to put the beforeeach function's commands into each test, but I'd like to avoid that if possible.  This'll be good for now to at least help out with cypress' flakiness as it is right now.

**Pull request merged**